### PR TITLE
height/width added https://github.com/EnsembleUI/ensemble/issues/238

### PR DESCRIPTION
**Pull request by Moeez1999**

### CODE REVIEW

File: lib/widget/button.dart [hunk 1]
```diff
@@ -95,9 +99,12 @@ class ButtonState extends WidgetState<Button> {
     }
 
     // add margin if specified
-    return widget._controller.margin != null ?
-      Padding(padding: widget._controller.margin!, child: rtn) :
-      rtn;
+    return Container(
```
Comment (vusters) [inline lib/widget/button.dart:102]: Hmm, Button should configure its own width/height and not rely on the Container outside. Same goes with border and such. The only thing outside the button is probably the margin, and for that use a Margin widget.
If you want to change the width/height, see ThemeUtils.getButtonStyle(). Inside you you can set width/height, minWidth/minHeight, maxwidth/maxHeight.

File: lib/util/theme_utils.dart [hunk 2]
```diff
@@ -34,13 +34,15 @@ class ThemeUtils {
     if (isOutline) {
       return TextButton.styleFrom(
         padding: padding,
+        fixedSize: Size(buttonWidth??Size.infinite.width, buttonHeight??Size.infinite.height),
```
Comment (vusters) [inline lib/util/theme_utils.dart:37]: This looks dangerous - setting infinite size. Did you try on all different permutations?

File: lib/util/theme_utils.dart [hunk 3]
```diff
@@ -22,7 +22,7 @@ class ThemeUtils {
 
   /// this function is also called while building the button, so make sure we don't use any fallback
   /// to ensure the style reverts to the button theming
-  static ButtonStyle getButtonStyle({required bool isOutline, Color? backgroundColor, Color? color, RoundedRectangleBorder? border, EdgeInsets? padding, FontWeight? fontWeight, int? fontSize}) {
+  static ButtonStyle getButtonStyle({required bool isOutline, Color? backgroundColor, Color? color, RoundedRectangleBorder? border, EdgeInsets? padding, FontWeight? fontWeight, int? fontSize,double? buttonWidth,double? buttonHeight}) {
```
Comment (vusters) [inline lib/util/theme_utils.dart:25]: use integer for width/height as that is what the user enters. You can convert to double when you need to `buttonWidth?.toDouble() ?? ...`

File: lib/util/theme_utils.dart (hunk 2 above)
Comment (vusters) [reply lib/util/theme_utils.dart:37]: put this Button (without explicit size) inside a Row or Column and it still work?

Comment (Moeez1999) [reply lib/util/theme_utils.dart:37]: I have tested it without giving height and width in column and row and works fine for me i am testing it with other use cases.

**Pull request merged**